include/ddc/chunk.hpp

```diff
@@ -128,14 +127,17 @@ class Chunk<ElementType, DiscreteDomain<DDims...>, Allocator>
      */
     Chunk& operator=(Chunk&& other)
     {
-        assert(this != &other);
-        if (this->m_internal_mdspan.data()) {
-            std::allocator_traits<Allocator>::deallocate(m_allocator, this->data(), this->size());
+        if (this == &other) {
+            return *this;
+        }
```

Comment:
Shall we really change the behavior here ? To me it is not an error to handle self assignment

Reply:
Well, either we handle it like that or we go back to the assert, but we need a big warning in the comment and I don't see the reason to do so

Reply:
I thought it was classically handled like that. Do default move assignment operators really have a self-check assignment ?

Reply:
Let's handle it but without promoting it.